test/common/common/inline_map_test.cc

```diff
@@ -0,0 +1,162 @@
+#include "source/common/common/inline_map.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+TEST(InlineMapWithZeroInlineKey, InlineMapWithZeroInlineKeyTest) {
+  InlineMapDescriptor<std::string> descriptor;
+  descriptor.finalize();
+
+  auto map = InlineMap<std::string, std::string>::create(descriptor);
+
+  // Insert keys.
+  for (size_t i = 0; i < 100; ++i) {
+    map->insert("key_" + std::to_string(i), "value_" + std::to_string(i));
+  }
+
+  // Insert duplicate keys will fail.
+  for (size_t i = 0; i < 100; ++i) {
+    EXPECT_FALSE(map->insert("key_" + std::to_string(i), "value_" + std::to_string(i)).second);
+  }
+
+  // Use operator[] to insert keys could overwrite existing keys.
+  for (size_t i = 0; i < 100; ++i) {
+    (*map)["key_" + std::to_string(i)] = "value_" + std::to_string(i) + "_new";
+  }
+
+  // Lookup keys.
+  for (size_t i = 0; i < 100; ++i) {
+    EXPECT_EQ(*map->lookup("key_" + std::to_string(i)), "value_" + std::to_string(i) + "_new");
+  }
+
+  // Lookup non-existing keys.
+  EXPECT_EQ(map->lookup("non_existing_key"), OptRef<std::string>{});
+
+  // Remove keys.
+  for (size_t i = 0; i < 100; ++i) {
+    map->remove("key_" + std::to_string(i));
+  }
+  map.reset();
+}
+
+TEST(InlineMapWith20InlineKey, InlineMapWith20InlineKeyTest) {
+  InlineMapDescriptor<std::string> descriptor;
+
+  std::vector<InlineMapDescriptor<std::string>::Handle> handles;
+
+  // Create 20 inline keys.
+  for (size_t i = 0; i < 20; ++i) {
+    handles.push_back(descriptor.addInlineKey("key_" + std::to_string(i)));
```

Comment:
can you add more tests for size() here at various stages of populating via handle?